fhir-path/src/main/java/com/ibm/fhir/path/evaluator/FHIRPathEvaluator.java

```diff
@@ -367,8 +367,10 @@ private void reset() {
             if (arguments.size() < 0 || arguments.size() > 1) {
                 throw unexpectedNumberOfArguments(arguments.size(), "exists");
             }
-            Collection<FHIRPathNode> nodes = arguments.isEmpty() ? getCurrentContext() : visit(arguments.get(0));
-            return !nodes.isEmpty() ? SINGLETON_TRUE : SINGLETON_FALSE;
+            if (arguments.isEmpty()) {
+                return !getCurrentContext().isEmpty() ? SINGLETON_TRUE : SINGLETON_FALSE;
+            }
+            return isTrue(visit(arguments.get(0))) ? SINGLETON_TRUE : SINGLETON_FALSE;
```

Comment:
So this is consistent with the `where` function?
What is the result if the argument is non-empty and non-boolean?

Reply:
it will return false if the argument evaluates to a non-empty / non-boolean

Reply:
i confirmed with john that this matches the `where` function, so this looks good to me